library/core/registry/src/main/java/org/quiltmc/qsl/registry/impl/event/DelayedRegistry.java

```diff
@@ -0,0 +1,243 @@
+		super(registry.getKey(), registry.getLifecycle());
+
+		this.wrapped = registry;
+
+		if (registry instanceof SimpleRegistryAccessor simpleRegistry) {
+			this.customHolderProvider = simpleRegistry.getCustomHolderProvider();
+		} else {
+			this.customHolderProvider = null;
+		}
+	}
+
+	@Override
+	public @Nullable Identifier getId(T entry) {
+		return this.wrapped.getId(entry);
+	}
+
+	@Override
+	public Optional<RegistryKey<T>> getKey(T entry) {
+		return this.wrapped.getKey(entry);
+	}
+
+	@Override
+	public int getRawId(@Nullable T entry) {
+		return this.wrapped.getRawId(entry);
+	}
+
+	@Override
+	public @Nullable T get(int index) {
+		return this.wrapped.get(index);
+	}
+
+	@Override
+	public int size() {
+		return this.wrapped.size();
+	}
+
+	@Override
+	public @Nullable T get(@Nullable RegistryKey<T> key) {
+		return this.wrapped.get(key);
+	}
+
+	@Override
+	public @Nullable T get(@Nullable Identifier id) {
+		return this.wrapped.get(id);
+	}
+
+	@Override
+	public Lifecycle getEntryLifecycle(T entry) {
+		return this.wrapped.getEntryLifecycle(entry);
+	}
+
+	@Override
+	public Lifecycle getLifecycle() {
+		return this.wrapped.getLifecycle();
+	}
+
+	@Override
+	public Set<Identifier> getIds() {
+		return this.wrapped.getIds();
+	}
+
+	@Override
+	public Set<Map.Entry<RegistryKey<T>, T>> getEntries() {
+		return this.wrapped.getEntries();
+	}
+
+	@Override
+	public Optional<Holder<T>> getRandom(Random random) {
+		return this.wrapped.getRandom(random);
+	}
+
+	@Override
+	public boolean containsId(Identifier id) {
+		return this.wrapped.containsId(id);
+	}
+
+	@Override
+	public boolean contains(RegistryKey<T> key) {
+		return this.wrapped.contains(key);
+	}
+
+	@Override
+	public Registry<T> freeze() {
+		// Refuse freezing.
+		return this;
+	}
+
+	@Override
+	public Holder<T> getOrCreateHolder(RegistryKey<T> key) {
+		return this.wrapped.getOrCreateHolder(key);
+	}
+
+	@Override
+	public Holder.Reference<T> createIntrusiveHolder(T holder) {
+		return this.wrapped.createIntrusiveHolder(holder);
+	}
+
+	@Override
+	public Optional<Holder<T>> getHolder(int index) {
+		return this.wrapped.getHolder(index);
+	}
+
+	@Override
+	public Optional<Holder<T>> getHolder(RegistryKey<T> key) {
+		return this.wrapped.getHolder(key);
+	}
+
+	@Override
+	public Stream<Holder.Reference<T>> holders() {
+		return this.wrapped.holders();
+	}
+
+	@Override
+	public Optional<HolderSet.NamedSet<T>> getTag(TagKey<T> tag) {
+		return this.wrapped.getTag(tag);
+	}
+
+	@Override
+	public HolderSet.NamedSet<T> getOrCreateTag(TagKey<T> key) {
+		return this.wrapped.getOrCreateTag(key);
+	}
+
+	@Override
+	public Stream<Pair<TagKey<T>, HolderSet.NamedSet<T>>> getTags() {
+		return this.wrapped.getTags();
+	}
+
+	@Override
+	public Stream<TagKey<T>> getTagKeys() {
+		return this.wrapped.getTagKeys();
+	}
+
+	@Override
+	public boolean isKnownTag(TagKey<T> tag) {
+		return this.wrapped.isKnownTag(tag);
+	}
+
+	@Override
+	public void resetTags() {
+		// noop
```

Comment:
Should this and `bindTags` throw a UOE like `set` and `replace` do?

Reply:
Probably